pkg/doc/did/doc.go

```diff
@@ -204,32 +212,77 @@ func ParseDocument(data []byte) (*Doc, error) {
 		return nil, err
 	}
 
+	doc := &Doc{
+		ID:      raw.ID,
+		Created: raw.Created,
+		Updated: raw.Updated,
+	}
+
 	context := raw.ParseContext()
+	doc.Context = context
+	doc.Service = populateServices(raw.Service)
 
 	publicKeys, err := populatePublicKeys(context[0], raw.PublicKey)
 	if err != nil {
 		return nil, fmt.Errorf("populate public keys failed: %w", err)
 	}
 
-	authPKs, err := populateAuthentications(context[0], raw.Authentication, publicKeys)
+	doc.PublicKey = publicKeys
+
+	err = populateVerificationRelationships(doc, raw, publicKeys)
+	if err != nil {
+		return nil, err
+	}
+
+	keyAgreements, err := populateKeyAgreements(raw.KeyAgreement)
 	if err != nil {
-		return nil, fmt.Errorf("populate authentications failed: %w", err)
+		return nil, fmt.Errorf("populate key agreements failed: %w", err)
 	}
 
+	doc.KeyAgreement = keyAgreements
+
 	proofs, err := populateProofs(context[0], raw.Proof)
 	if err != nil {
 		return nil, fmt.Errorf("populate proofs failed: %w", err)
 	}
 
-	return &Doc{Context: context,
-		ID:             raw.ID,
-		PublicKey:      publicKeys,
-		Service:        populateServices(raw.Service),
-		Authentication: authPKs,
-		Created:        raw.Created,
-		Updated:        raw.Updated,
-		Proof:          proofs,
-	}, nil
+	doc.Proof = proofs
+
+	return doc, nil
+}
+
+func populateVerificationRelationships(doc *Doc, raw *rawDoc, pks []PublicKey) error {
+	context := doc.Context[0]
+
+	authentications, err := populateVerificationMethods(context, raw.Authentication, pks)
+	if err != nil {
+		return fmt.Errorf("populate authentications failed: %w", err)
```

Comment:
I noticed that it's a common pattern to use the "failed" keyword at error wrapping here.
But according to [Uber Go Style Guide](https://github.com/uber-go/guide/blob/master/style.md#error-wrapping), it's better to 
> When adding context to returned errors, keep the context succinct by avoiding phrases like "failed to", which state the obvious and pile up as the error percolates up through the stack